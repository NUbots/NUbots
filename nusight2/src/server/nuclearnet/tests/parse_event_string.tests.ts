import { describe, expect, it } from "vitest";

import { parseEventString } from "../parse_event_string";

describe.only("parseEventString()", () => {
  it("parses event without subtype", () => {
    expect(parseEventString("my.type")).toStrictEqual({
      type: "my.type",
      subtype: undefined,
    });

    expect(parseEventString("my.type#")).toStrictEqual({
      type: "my.type",
      subtype: undefined,
    });
  });

  it("parses event with valid subtype", () => {
    expect(parseEventString("my.type#0")).toStrictEqual({
      type: "my.type",
      subtype: 0,
    });

    expect(parseEventString("my.type#20")).toStrictEqual({
      type: "my.type",
      subtype: 20,
    });
  });

  it("throws for event with invalid subtype by default", () => {
    expect(() => {
      parseEventString("my.type#not-a-number");
    }).toThrowErrorMatchingSnapshot();
  });

  it("returns empty subtype for event with invalid subtype when configured to not throw", () => {
    expect(parseEventString("my.type#not-a-number", { throwOnInvalidSubtype: false })).toStrictEqual({
      type: "my.type",
      subtype: undefined,
    });
  });
});
