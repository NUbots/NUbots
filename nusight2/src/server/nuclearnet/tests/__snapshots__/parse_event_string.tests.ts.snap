// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseEventString() > throws for event with invalid subtype by default 1`] = `[Error: invalid subtype in event "my.type#not-a-number". subtype "not-a-number" is not a number]`;
